src/interface/obsidian/src/utils.ts

```diff
@@ -48,14 +48,29 @@ function filenameToMimeType (filename: TFile): string {
     }
 }
 
-export const supportedImageFilesTypes = ['png', 'jpg', 'jpeg'];
-export const supportedBinaryFileTypes = ['pdf'].concat(supportedImageFilesTypes);
-export const supportedFileTypes = ['md', 'markdown'].concat(supportedBinaryFileTypes);
-
-export async function updateContentIndex(vault: Vault, setting: KhojSetting, lastSync: Map<TFile, number>, regenerate: boolean = false): Promise<Map<TFile, number>> {
+export const fileTypeToExtension = {
+    'pdf': ['pdf'],
+    'image': ['png', 'jpg', 'jpeg'],
+    'markdown': ['md', 'markdown'],
+};
+export const supportedImageFilesTypes = fileTypeToExtension.image;
+export const supportedBinaryFileTypes = fileTypeToExtension.pdf.concat(supportedImageFilesTypes);
+export const supportedFileTypes = fileTypeToExtension.markdown.concat(supportedBinaryFileTypes);
```

Comment:
Where is the user setting being applied? It seems like it's just using all the allowed file types and not filtering anything out.

Reply:
Ah nevermind, it's happening below with the use of the `KhojSetting` object.